labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise02.scala

```diff
@@ -39,29 +47,9 @@ object ListManipulationExercise02 {
    * in a one-liner.
    */
   def separateTheYoungFromTheOld(persons: List[Person]): List[List[String]] = {
-    var youngins: ListBuffer[Person] = new ListBuffer[Person]()
-    var elders: ListBuffer[Person] = new ListBuffer[Person]()
-    var validYoungNames: ListBuffer[String] = new ListBuffer[String]()
-    var validOldNames: ListBuffer[String] = new ListBuffer[String]()
-
-    for (person <- persons) {
-        if (person.age < 18) {
-          youngins += person
-        } else {
-          elders += person
-        }
-    }
-
-    var sortedYoung = youngins.toList.sortBy(_.age)
-    var sortedOld = elders.toList.sortBy(_.age)
-
-    for (young <- sortedYoung) {
-      validYoungNames += young.firstName
-    }
-    for (old <- sortedOld) {
-      validOldNames += old.firstName
-    }
-    List(validYoungNames.toList, validOldNames.toList)
+    // One-Liners-R-Us
+    List(persons.filter((A)=> A.age < 18).sortBy(_.age).map((A)=> A.firstName)) ++ List(persons.filter((A)=> A.age >= 18).sortBy(_.age).map((A)=> A.firstName))
```

Comment:
I had the exact one-liner! I'm sure neither you nor I would actually code like this considering how hard it is to read unless we were being egged on by the comments in the lab. The provided solution was pretty elegant and readable since they split out the filter and sortBy to individual methods inside the class.